### ISSUE TITLE
This code is interpretted as a string/char for some reason.

### ISSUE THREAD

**Issue by sol-vin** [crystal-lang-tools/vscode-crystal-lang#204]
**Describe the bug**
For some reason the following code creates a visual bug with the highlighting.

**To Reproduce**
1. Type `ty = (tile.index/width).to_i * tile_height`
2. Add a space like `ty = (tile.index /width).to_i * tile_height`
3. Highlighting is broken

**Expected behavior**
Highlighting should not break.

**Piece of Code**
[A piece of code to reproduce the problem](https://gist.github.com/sol-vin/98fde9f10a1bedcbe1b8441ed0b0186b)

**Screenshots**
Video:
https://github.com/user-attachments/assets/2bd8d707-9ba1-4198-aa0a-c96a38fd885d

**Desktop (please complete the following information):**
 - OS: Windows
 - Crystal version: 1.13.2
 - This plugin version: v0.9.6

**Additional context**
None, other than changing the method name to anything other than index fixes it?

**Comment by nobodywasishere** [comment]
It's probably due to it considering that `/` to be the start of a regex and not being able to find the end. If you put a `# /` at the end of that line, is the error contained to that line? I need to go through and re-do the syntax highlighting rules, or just figure out a way to do syntax highlighting in vscode with tree-sitter and do that instead (the latter being a lot less error prone, but harder to write)

**Comment by sol-vin** [comment]
Yep, highlighting unbreaks when commented out. However, this stops breaking when the method name `index` is changed to anything else

**Comment by sol-vin** [comment]
https://github.com/user-attachments/assets/7f3786e8-98d8-4121-b294-8f449de2838e

**Comment by nobodywasishere** [comment]
Using tree sitter does look possible, just requires the use of the semantic tokens API
https://marketplace.visualstudio.com/items?itemName=AlecGhost.tree-sitter-vscode
